airflow/models.py

```diff
@@ -2630,6 +2630,65 @@ def xcom_pull(
             include_prior_dates=include_prior_dates)
 
 
+class ExecutorQueue(Base):
+    __tablename__ = "executor_queue"
+
+    key = Column(String(ID_LEN), primary_key=True)
+    id = Column(String(ID_LEN), nullable=False)
+    updated = Column(DateTime, nullable=False)
+
+
+class ExecutorQueueManager(dict):
+    def __init__(self, *args, **kwargs):
+        super(ExecutorQueueManager, self).__init__(*args, **kwargs)
+        self.session = settings.Session()
+
+    def __getitem__(self, key):
+        qry = (self.session.query(ExecutorQueue)
```

Comment:
Will these queries be expensive?

Reply:
No, key value lookup. No scans, nothing should be faster than this.

Reply:
Still has roundtrip time